lib/impl/testing.dart

```diff
@@ -83,6 +95,88 @@ class TestManager implements Disposable {
     runTestFile(path, allowWithoutTestName: true);
   }
 
+  /// Finds the test file most closely associted with this test. If the given
+  /// path is a test file itself, or not associated test file is found, the
+  /// original path is returned.
+  String _findAssociatedTest(DartProject project, String path) {
+    if (path == null || path.endsWith('_test.dart')) return path;
+
+    String prefix = project.path;
+    String pathFragment = path.substring(prefix.length + 1);
+
+    for (String fragment in getPossibleTestPaths(pathFragment, separator: fs.separator)) {
+      String testPath = fs.join(prefix, fragment);
+      if (fs.existsSync(testPath)) {
+        return testPath;
+      }
+    }
+
+    return path;
+  }
+
+  Future _createTest([AtomEvent _]) async {
+    String path = atom.workspace.getActiveTextEditor()?.getPath();
+    if (path == null) {
+      atom.notifications.addInfo('No active editor.');
+      return;
+    }
+
+    DartProject project = projectManager.getProjectFor(path);
+    if (project == null) {
+      atom.notifications.addInfo('No active Dart project.');
+      return;
+    }
+
+    if (!isDartFile(path)) {
+      atom.notifications.addInfo('Current editor is not a Dart file.', description: path);
+      return;
+    }
+
+    String pathFragment = project.getRelative(path);
+
+    if (!pathFragment.startsWith('lib${fs.separator}')) {
+      atom.notifications.addInfo("This action requires the file to be in the 'lib' folder.");
+      return;
+    }
+
+    // Remove lib/.
+    pathFragment = pathFragment.substring(4);
+    String testPath = pathFragment.substring(0, pathFragment.length - 5) + '_test.dart';
+    testPath = fs.join(project.path, 'test', testPath);
+
+    if (fs.existsSync(testPath)) {
+      atom.workspace.open(testPath);
+      return;
+    }
+
+    String packageName = project.getSelfRefName();
+    String groupName = fs.basename(pathFragment);
+    groupName = groupName.substring(0, groupName.length - 5);
+
+    File file = new File.fromPath(testPath);
+    file.writeSync('''
+import 'package:${packageName}/${pathFragment}';
+import 'package:test/test.dart';
+
+main() => defineTests();
+
+defineTests() {
+  group('${groupName}', () {
+    test('todo', () {
+      // TODO: Implement test.
+
+    });
+  });
+}
+''');
+
+    await new Future.delayed(Duration.ZERO);
+    atom.workspace.open(testPath);
+
+    await new Future.delayed(Duration.ZERO);
```

Comment:
Updated to wait on the editor being opened from above (so we don't have to have an arbitrary delay).